### ISSUE TITLE
Overflow Error

### ISSUE THREAD

**Issue by lulinxuan** [kevinjliang/tf-Faster-RCNN#18]
./Lib/bbox_transform.py:66: RuntimeWarning: overflow encountered in exp                                                                                                             | 0/5000 [00:00<?, ?it/s]
  pred_w = np.exp(dw) * widths[:, np.newaxis]
../Lib/bbox_transform.py:67: RuntimeWarning: overflow encountered in exp
  pred_h = np.exp(dh) * heights[:, np.newaxis]
                                                                                                                                                                                                            ../Lib/bbox_transform.py:67: RuntimeWarning: overflow encountered in multiply                                                                                             | 311/5000 [00:00<00:13, 346.46it/s]
  pred_h = np.exp(dh) * heights[:, np.newaxis]
                                                                                                                                                                                                            ../Lib/bbox_transform.py:66: RuntimeWarning: overflow encountered in multiply                                                                                             | 347/5000 [00:01<00:13, 348.99it/s]
  pred_w = np.exp(dw) * widths[:, np.newaxis]


I followed the training steps, and when I run 
python faster_rcnn_conv5.py -r 1 -m [Model num] -f [epoch to restore] -t 0 -i 1
I got the problem. I print the dw and dh values, they are +-500000, which does not seem to be correct.

**Comment by kevinjliang** [comment]
Did you run `python faster_rcnn_conv5.py -r 1 -m [Model num] -f [epoch to restore] -t 0 -i 1` exactly, or did you fill in numbers for [Model num] and [epoch to restore]? If it's the former, that's a problem.

As for the overflow, that's not something I've encountered before. I'll look into it when I get the chance.

**Comment by lulinxuan** [comment]
Yeah of course I fill in numbers.

**Comment by kevinjliang** [comment]
Were you able to train a model properly and save some checkpoints?

**Comment by lulinxuan** [comment]
Yes I can. I got a ckpt file. Actually the output images for the eval have quite nice bbox, but the accuracy displayed is 0. I'm using python 2.7 and tensorflow 1.0.

**Comment by kevinjliang** [comment]
Okay, that was going to be next question. Unfortunately, I didn't write this code to be 2.7 compatible. That said, I included instructions for installing Python 3 with Anaconda; if you follow them, you can have Python 3.5+ on your system in an isolated environment that won't interfere with your system Python version.

**Comment by lulinxuan** [comment]
May I know why you use np.exp(dw) instead of just dw? Since dw is delta_w, why you calculate exp(dw)? It does not make sense.

**Comment by kevinjliang** [comment]
That's the way the bounding box regression to the anchor boxes has been parameterized. The model predicts the height and width delta in terms of a ratio to make it more scale invariant; these ratios must be positive, but the typical neural network produces real values, so we instead predict the log ratio. The exponential "undoes" the log to recover the ratio.

**Comment by lulinxuan** [comment]
OK thanks! I finally got the problem. It's caused by batch_norm, when the is_training is set to false, it will produce wrong value. I tried to set it to true through the eval, it works well.

**Comment by kevinjliang** [comment]
Good to hear it's working for you!
